credoai/artifacts/data/tabular_data.py

```diff
@@ -72,25 +76,30 @@ def _process_y(self, y):
                 y = pd_type(y, name="target")
         return y
 
+    def _validate_X(self):
+        try:
+            pd.DataFrame(self.X)
```

Comment:
Technically someone could pass a dictionary and this wouldn't fail. They'd be violating the documentation but I _think_ it'd still end up in the right format based on other validation. Nevertheless, it might be better to explicitly forbid it through `isinstance` checking since there could be unanticipated issues with dictionaries.

Example:
```
>>> blarg = {'col1': [4, 5, 6], 
... 'col2': [1, 2, 3]}
>>> import pandas as pd
>>> pd.DataFrame(blarg)
   col1  col2
0     4     1
1     5     2
2     6     3
```

Reply:
I looked for validation of "array-like" as requested by sklearn. It doesn't seem like there is such validation. This seems like quite an odd case to be passed and not revealed by other validations